e2e/internal/e2e/keyexec.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package e2e
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func getImportScript(kpath string) string {
+	// Yes, this uses /usr/bin/expect
+	return fmt.Sprintf(`
+set timeout -1
+
+set psk "e2etests"
+
+spawn singularity key import %s
+
+expect "Enter your old password : "
+send "${psk}\r"
+
+expect "Enter a new password for this key : "
+send "${psk}\r"
+
+expect "Retype your passphrase : "
+send "${psk}\r"
+
+expect eof
+`, kpath)
+}
+
+func getExportScript(num int, kpath, armor, psk string) string {
+	// Yes, this uses /usr/bin/expect
+	return fmt.Sprintf(`
+set timeout -1
+
+spawn singularity key export --secret %s %s
+
+expect "Enter # of signing key to use : "
+send "%v\r"
+
+expect "Enter key passphrase : "
+send "%s\r"
+
+expect eof
+`, armor, kpath, num, psk)
+}
+
+var backupSypgp = filepath.Join(HomeDir(), ".singularity/sypgp/secret-keyring-backup")
```

Comment:
there's an exported function in the sypgp package that returns "~/.singularity/sypgp".